### PR TITLE
Changing OtherCourseSettings course_id to course FK.

### PR DESCRIPTION
**Pull request by ivanvgh**
### Description:

This PR changes OtherCOurseSettings table, `course_id` CourseKeyField to course `CourseOverview` FK.
This decision was taken to have capabilities to make efficient queries against this table.

Btw, this change does not affect the custom management commands implemented to migrate/update courses to this table.

PS: The migration file was replaced by a new migration due to conflicts given on migrating with default values, this occurs because the column that is being changed to a FK was a PK, and that is why `id` as default PK is created.

### CODE REVIEW

File: openedx_external_enrollments/edxapp_wrapper/get_course_mode.py [hunk 1]
```diff
@@ -10,3 +10,11 @@ def get_course_mode():
     backend = import_module(backend_function)
 
     return backend.get_course_mode()
+
+
+def get_course_overview():
```
Comment (Squirrel18) [inline openedx_external_enrollments/edxapp_wrapper/get_course_mode.py:15]: This definition must be included in a course module file. @ivanvgh

File: openedx_external_enrollments/edxapp_wrapper/get_course_mode.py [hunk 2]
```diff
@@ -10,3 +10,11 @@ def get_course_mode():
...
+    """ Get get_course_overview Method. """
```
Comment (Squirrel18) [inline openedx_external_enrollments/edxapp_wrapper/get_course_mode.py:16]:
```suggestion
    """ Get get_course_overview method. """
```

File: openedx_external_enrollments/settings/test.py [hunk 3]
```diff
@@ -6,6 +6,19 @@
...
+class DisableMigrations(dict):
+    """Helper class that disable migrations."""
+    def __contains__(self, item):
+        return True
+
+    def __getitem__(self, item):
+        return None
+
+
+# Disables migrations due to OtherCourseSettings table migration depends from an out of context migration file.
```
Comment (Squirrel18) [inline openedx_external_enrollments/settings/test.py:19]:
```suggestion
# Disables migrations due to OtherCourseSettings table migration depends on an edx-platform migration file.
```

File: openedx_external_enrollments/settings/test.py [hunk 4]
```diff
@@ -6,6 +6,19 @@
...
+MIGRATION_MODULES = DisableMigrations()
+
```
Comment (Squirrel18) [inline openedx_external_enrollments/settings/test.py:21]: Unnecessary blank line. @ivanvgh

File: openedx_external_enrollments/settings/test.py [hunk 5]
```diff
@@ -6,6 +6,19 @@
 
 from .common import *  # pylint: disable=wildcard-import
 
+
+class DisableMigrations(dict):
```
Comment (Squirrel18) [inline openedx_external_enrollments/settings/test.py:10]: I think you don't need to add this. Ref: https://docs.djangoproject.com/en/2.2/ref/settings/#migration-modules @ivanvgh

File: openedx_external_enrollments/tests/tests_backends.py [hunk 6]
```diff
@@ -34,3 +39,8 @@ def get_enrollment_track_updated_signal():
 def get_unenroll_done_signal():
     """Test get_unenroll_done_signal method."""
     return Mock()
+
+
+def get_course_overview():
+    """Test get_unenroll_done_signal method."""
```
Comment (Squirrel18) [inline openedx_external_enrollments/tests/tests_backends.py:45]: Please update the doc-string. @ivanvgh

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/settings/test.py (hunk 5 above)
Comment (ivanvgh) [reply openedx_external_enrollments/settings/test.py:10]: Nice catch!

**Review by diegomillan — DISMISSED** [review]
Please handle carefully the migration steps, including command rerunning @ivanvgh,  also let @Alec4r all the necessary to make it work (at DB level).

**Review by Squirrel18 — APPROVED** [review]

**Review by diegomillan — APPROVED** [review]

**Pull request merged**